src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.State.cs

```diff
@@ -0,0 +1,297 @@
+                        // In case there's no namespace declaration in the document, we used an empty span as key, 
+                        // since a valid namespace declaration node can't have zero length.
+                        var span = namespaceDeclaration?.Span ?? new TextSpan(0, 0);
+                        if (shouldTrigger)
+                        {
+                            spansForNamespaceDeclaration[span] = namespaceDeclaration;
+                        }
+                        else
+                        {
+                            return (false, null, null);
+                        }
+                    }
+
+                    if (spansForNamespaceDeclaration.Count == 1)
+                    {
+                        var namespaceDecl = spansForNamespaceDeclaration.Values.Single();
+                        var declaredNamespace = namespaceDecl == null
+                            // namespaceDecl == null means the target namespace is global namespace.
+                            ? string.Empty
+                            // Since the node in each document has identical type and span, 
+                            // they should have same name.
+                            : SyntaxGenerator.GetGenerator(documents.First()).GetName(namespaceDecl);
+
+                        return (true, defaultNamespace, declaredNamespace);
+                    }
+                }
+
+                return (false, null, null);
+            }
+
+            public static async Task<State> CreateAsync(
+                AbstractSyncNamespaceService<TNamespaceDeclarationSyntax, TCompilationUnitSyntax> service, 
+                Document document, 
+                TextSpan textSpan, 
+                CancellationToken cancellationToken)
+            {
+                if (document.Project.FilePath == null
+                    || !textSpan.IsEmpty
+                    || document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles
+                    || document.IsGeneratedCode(cancellationToken))
+                {
+                    return null;
+                }
+
+                if (!IsSupportedLinkedDocument(document, out var documentIds))
+                {
+                    return null;
+                }
+
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var solution = document.Project.Solution;
+                var documents = documentIds.Select(id => solution.GetDocument(id)).ToImmutableArray();
+
+                var result = await TryGetDefaultNamespaceAndNamespaceDeclarationAsync(
+                    documents, service, syntaxFacts, textSpan, cancellationToken).ConfigureAwait(false);
+
+                if (!result.shouldTriggerRefactoring)
+                {
+                    return null;
+                }
+
+                var defaultNamespace = result.defaultNamespace;
+                var declaredNamespace = result.declaredNamespace;
```

Comment:
you don't actually use 'result'.  so just deconstruct where you declare the variable.